lib/nil.rb

```diff
@@ -0,0 +1,6 @@
+class NilClass
+ def to_hash
+   to_h
+ end
+end
+data = {test: :t}.merge!(nil)
```

Comment:
I respect your ingenuity! However, I think you are overcomplicating things - if you are getting a nil value error, figure out a way to rescue the error or to prevent it from occuring, rather than monkeypatching extra methods onto NilClass.